code/game/machinery/scan_gate.dm

```diff
@@ -160,11 +177,26 @@
 				if(H.nutrition >= detect_nutrition && detect_nutrition == NUTRITION_LEVEL_FAT)
 					beep = TRUE
 
+	if(!ignore_signals)
+		color = wires.get_color_of_wire(WIRE_PROCEED)
+		var/obj/item/assembly/S = wires.get_attached(color)

```

Comment:
You do these 3 lines 3 times in 3 different places. Just do them once at the end.

Reply:
I'll condense this - it made more sense when there was only 2.